### PR TITLE
Fix for Swift 4.2

### PR DESCRIPTION
**Pull request by shaqaruden**
Only issue I have found is. Camera either freezes or is black when first starting the app. Go to home and open again and it works fine. Tested on iPhone X

### CODE REVIEW

File: Sources/Controllers/CameraViewController.swift [hunk 1]
```diff
@@ -66,7 +66,7 @@ public final class CameraViewController: UIViewController {
         captureDevice.unlockForConfiguration()
       } catch {}
 
-      flashButton.setImage(torchMode.image, for: UIControlState())
+        flashButton.setImage(torchMode.image, for: UIControl.State())
```
Comment (vadymmarkov) [inline Sources/Controllers/CameraViewController.swift]: Could you remove this extra space?

File: Sources/Controllers/MessageViewController.swift [hunk 2]
```diff
@@ -58,7 +58,7 @@ public final class MessageViewController: UIViewController {
 
    - Parameter style: The current blur style.
    */
-  private func animate(blurStyle: UIBlurEffectStyle) {
+    private func animate(blurStyle: UIBlurEffect.Style) {
```
Comment (vadymmarkov) [inline Sources/Controllers/MessageViewController.swift]: There is an extra space here.

File: Sources/Controllers/BarcodeScannerViewController.swift [hunk 3]
```diff
@@ -303,7 +303,7 @@ extension BarcodeScannerViewController: CameraViewControllerDelegate {
 
   func cameraViewControllerDidTapSettingsButton(_ controller: CameraViewController) {
     DispatchQueue.main.async {
-      if let settingsURL = URL(string: UIApplicationOpenSettingsURLString) {
+        if let settingsURL = URL(string: UIApplication.openSettingsURLString) {
```
Comment (vadymmarkov) [inline Sources/Controllers/BarcodeScannerViewController.swift]: There is an extra space here.

File: Sources/Controllers/CameraViewController.swift [hunk 4]
```diff
@@ -173,7 +173,7 @@ public final class CameraViewController: UIViewController {
     NotificationCenter.default.addObserver(
       self,
       selector: #selector(appWillEnterForeground),
-      name: NSNotification.Name.UIApplicationWillEnterForeground,
+      name: NSNotification.Name.NSExtensionHostWillEnterForeground,
```
Comment (vadymmarkov) [inline Sources/Controllers/CameraViewController.swift]: Shouldn't it become `UIApplication.willEnterForegroundNotification`?

**Comment by Kalvin126** [comment]
Any update on this?

File: Sources/Controllers/CameraViewController.swift (hunk 1 above)
Comment (shaqaruden) [reply Sources/Controllers/CameraViewController.swift]: Fixed

File: Sources/Controllers/MessageViewController.swift (hunk 2 above)
Comment (shaqaruden) [reply Sources/Controllers/MessageViewController.swift]: Fixed

File: Sources/Controllers/BarcodeScannerViewController.swift (hunk 3 above)
Comment (shaqaruden) [reply Sources/Controllers/BarcodeScannerViewController.swift]: Fixed

File: Sources/Controllers/CameraViewController.swift (hunk 4 above)
Comment (shaqaruden) [reply Sources/Controllers/CameraViewController.swift]: Your right, updated

**Comment by shaqaruden** [comment]
Sorry for not getting back to this sooner, life happened. I made changes based on feedback. As for the failed check, the code is correct. Xcode changed it to UIBlurEffect.Style during the conversion and complains otherwise.

Image of Xcode complaining

I have checked other project that implemented UIBlurEffectStyle and they also switched to UIBlurEffect.Style when moving to Swift 4.2

**Comment by shaqaruden** [comment]
Travis now passes

**Comment by shaqaruden** [comment]
This is my first time contributing to a swift repo. Sorry for the delay.

**Comment by shaqaruden** [comment]
@vadymmarkov This is ready to be merged. Travis now passes

**Comment by 3lvis** [comment]
Fixed already, thank you!

**Pull request closed without merging**